common/src/Microsoft.Azure.IIoT.Serializers.NewtonSoft/src/Default/NewtonSoftJsonSerializer.cs

```diff
@@ -72,21 +74,24 @@ public class NewtonSoftJsonSerializer : IJsonSerializerSettingsProvider,
             try {
                 // TODO move to .net 3 to use readonly span as stream source
                 var jsonSerializer = JsonSerializer.CreateDefault(Settings);
-                using (var stream = new MemoryStream(buffer.ToArray()))
+                var bufferArray = buffer.ToArray();
+                using (var stream = new MemoryStream(bufferArray))
                 using (var reader = new StreamReader(stream, ContentEncoding)) {
 
                     // Validate json if schema is provided.
                     if (schemaReader != null) {
-                        var messages = new List<string>();
-                        using (var jsonSchemaReader = new JsonTextReader(schemaReader)) {
-                            var schema = JSchema.Load(jsonSchemaReader);
-                            using (var validatingReader = new JSchemaValidatingReader(new JsonTextReader(reader))) {
-                                validatingReader.Schema = schema;
-                                validatingReader.ValidationEventHandler += (o, a) => messages.Add(a.Message);
-                                var serializedType = jsonSerializer.Deserialize(validatingReader, type);
-
-                                return messages.Count > 0 ? throw new JsonSerializationException(string.Join(", ", messages.ToArray())) : serializedType;
-                            }
+                        // the inbound data 
+                        using (var validationDataStream = new MemoryStream(bufferArray))
+                        using (var validationReader = new StreamReader(validationDataStream, ContentEncoding)) {
+                            
+                            var schema = Task.Run(() => JsonSchema.FromJsonAsync(schemaReader.ReadToEnd())).GetAwaiter().GetResult();
```

Comment:
Wonder if we need Task.Run() here, could we simplify it i.e. "JsonSchema.FromJsonAsync(schemaReader.ReadToEnd()).GetAwaiter().GetResult();"?

Reply:
As an FYI: https://stackoverflow.com/a/59019473/1276028